File: src/main/java/org/indymon/service/object/Check.java
package org.indymon.service.object;

import org.indymon.common.model.CheckDefinition;
import org.indymon.common.model.CheckStatus;

public class Check {

    private CheckDefinition definition;
    private CheckStatus status;

    public Check(CheckDefinition definition) {
        this.definition = definition;
    }

    public CheckDefinition getDefinition() {
        return definition;
    }

    public void setDefinition(CheckDefinition definition) {
        this.definition = definition;
    }

    public CheckStatus getStatus() {
        return status;
    }

    public void setStatus(CheckStatus status) {
        this.status = status;
    }
}

File: static/js/Graph/Object/Check.js
"use strict";

(function ($) {

    IndyMon.Graph.Object.Check = Backbone.Model.extend({

        initialize: function (params) {
            this.definition = params.definition;
            this.reset();
        },

        reset: function () {
            this.status = 0;
            this.message = "?";
        },

        getSymbol: function () {
            return this.definition.symbol;
        },

        getStatus: function () {
            return this.status;
        },

        getMessage: function () {
            return this.message;
        },

        setNewStatus: function (newStatus) {
            this.status = newStatus.status;
            this.message = newStatus.message;
        }

    });

})(jQuery);
